eogrow/pipelines/split_grid.py

```diff
@@ -0,0 +1,184 @@
+        if self.config.buffer != "auto":
+            return self.config.buffer
+
+        area_config = self.area_manager.config
+        if isinstance(area_config, UtmZoneAreaManager.Schema):
+            return area_config.patch_buffer_x, area_config.patch_buffer_y
+        if isinstance(area_config, BatchAreaManager.Schema):
+            res = area_config.resolution
+            return area_config.tile_buffer_x * res, area_config.tile_buffer_y * res
+        raise ValueError(
+            f"Cannot infer buffer from area manager `{type(self.area_manager)}`. Please set the `buffer` parameter."
+        )
+
+    def build_workflow(self) -> EOWorkflow:
+        features = self._get_features()
+
+        input_path = self.storage.get_folder(self.config.input_folder_key)
+        load_node = EONode(LoadTask(input_path, filesystem=self.storage.filesystem, features=features))
+
+        processing_nodes = []
+        output_path = self.storage.get_folder(self.config.eopatch_output_folder_key)
+        for _ in range(self.config.split_x * self.config.split_y):
+            slice_task = SpatialSliceTask(features, raise_misaligned=self.config.raise_misaligned)
+            slice_node = EONode(slice_task, inputs=[load_node])
+
+            save_task = SaveTask(
+                output_path,
+                filesystem=self.storage.filesystem,
+                features=features,
+                overwrite_permission=OverwritePermission.OVERWRITE_FEATURES,
+            )
+            save_node = EONode(save_task, inputs=[slice_node])
+            processing_nodes.append(save_node)
+
+        return EOWorkflow.from_endnodes(*processing_nodes)
+
+    def get_execution_arguments(  # type: ignore[override]
+        self, workflow: EOWorkflow, bbox_splits: List[Tuple[NamedBBox, List[NamedBBox]]]
+    ) -> List[Dict[EONode, Dict[str, object]]]:
+        nodes = workflow.get_nodes()
+        load_node = nodes[0]
+        save_nodes = [node for node in nodes if isinstance(node.task, SaveTask)]
+        slice_nodes = [save_node.inputs[0] for save_node in save_nodes]
+
+        exec_args: List[Dict[EONode, Dict[str, object]]] = []
+        for (orig_name, _), split_bboxes in bbox_splits:
+            single_exec: Dict[EONode, Dict[str, object]] = {load_node: dict(eopatch_folder=orig_name)}
+
+            for i, (subbox_name, subbox) in enumerate(split_bboxes):
+                single_exec[slice_nodes[i]] = dict(bbox=subbox)
+                single_exec[save_nodes[i]] = dict(eopatch_folder=subbox_name)
+
+            exec_args.append(single_exec)
+
+        return exec_args
+
+    def _get_features(self) -> List[FeatureSpec]:
+        """Provides features that will be transformed by the pipeline."""
+        meta_features = [FeatureType.BBOX]
+        if any(f_type.is_temporal() for f_type, _ in self.config.features):
+            meta_features += [FeatureType.TIMESTAMP]
+
+        return self.config.features + meta_features
```

Comment:
This seems quite generic, perhaps could go into utils or something?

Reply:
i'll open an issue with this suggestion